common/util/bbb.ts

```diff
@@ -146,3 +156,157 @@ export class BBBMeeting {
+                logToUpdate.attendedTime += courseAttendanceLogInterval;
+                await entityManager.save(CourseAttendanceLog, logToUpdate);
+                await transactionLog.log(new CreateCourseAttendanceLogEvent(pupil, logToUpdate));
+                logger.info("Successfully updated log with id: ", logToUpdate.id);
+            } else {
+                try {
+                    // Create new log
+                    courseAttendanceLog.ip = ip;
+                    courseAttendanceLog.pupil = pupil;
+                    courseAttendanceLog.lecture = activeLecture;
+                    courseAttendanceLog.attendedTime = null;
+                    await entityManager.save(CourseAttendanceLog, courseAttendanceLog);
+                    await transactionLog.log(new CreateCourseAttendanceLogEvent(pupil, courseAttendanceLog));
+                    logger.info("Successfully saved new Course Attendance to lecture with id ", activeLecture.id);
+                } catch (e) {
+                    logger.error("Can't save new course attendance: " + e.message);
+                    logger.debug(courseAttendanceLog, e);
+                }
+            }
+        } else {
+            logger.error("Can't save new course attendance: no active lecture");
+        }
+    }
+}
+
+export async function getBBBMeetingAttendees(meetingID: string): Promise<Attendee[]> {
+    const callName = "getMeetingInfo";
+    const queryParams = encodeURI(`meetingID=${meetingID}`);
+
+    try {
+        const response = await axios.get(`${baseUrl}${callName}?${queryParams}&checksum=${hashToken(callName + queryParams + sharedSecret, "sha1")}`);
+        const jsonResponse = await parser.parseStringPromise(response.data);
+        return mapJSONtoAttendees(jsonResponse);
+    } catch (error) {
+        logger.debug(error);
+        return null;
+    }
+}
+
+export async function handleBBBMeetingInfos() {
+    const entityManager = getManager();
+    const meetings = await getBBBMeetings();
+    if (meetings != null) {
+        for (const meeting of meetings) {
+            const meetingAttendees = await getBBBMeetingAttendees(meeting.meetingID);
+            if (meetingAttendees != null) {
+                const map = new Map();
+                const filteredMeetingAttendees: Attendee[] = [];
+                for (const attendee of meetingAttendees) {
+                    if (!map.has(attendee.wix_id)) {
+                        map.set(attendee.wix_id, true);
+                        filteredMeetingAttendees.push(attendee);
+                    }
+                }
+                for (const attendee of filteredMeetingAttendees) {
+                    if (attendee.role && attendee.role === "VIEWER") {
+                        const pupilFromDB = await entityManager.findOne(Pupil, {wix_id: attendee.wix_id});
+                        if (pupilFromDB) {
+                            createOrUpdateCourseAttendanceLog(pupilFromDB, null, meeting.meetingID);
```

Comment:
Calling this with `meeting.meetingID` will only work if `meetingID` is actually the subcourse ID, not the course ID, I think. But if we look at https://github.com/corona-school/backend/blob/b5f2e4c0e6b92f0a7477f5799a99b24221d2b296/web/controllers/courseController/index.ts#L2392 this is set to the course ID when creating the BBB-meeting for the course. Is that correct? 
I just did a last test before going live and encountered exactly the problem that the course attendance log wasn't updated if the course ID is not equal to the subcourse ID. We should probably fix the meetingID to be set to the subcourse ID.

Reply:
Yes, you are right. I thought I did this change, but I did it in the other branch.
I will fix this.